src/engine/effects/engineeffectsmanager.cpp

```diff
@@ -201,79 +197,82 @@ void EngineEffectsManager::processInner(
         CSAMPLE* pIntermediateInput = m_buffer1.data();
         if (oldGain == CSAMPLE_GAIN_ONE && newGain == CSAMPLE_GAIN_ONE) {
-            // Avoid an unnecessary copy. EngineEffectRack::process does not modify the
+            // Avoid an unnecessary copy. EngineEffectChain::process does not modify the
             // input buffer when its input & output buffers are different, so this is okay.
             pIntermediateInput = pIn;
         } else {
-            SampleUtil::copyWithRampingGain(pIntermediateInput, pIn,
-                                            oldGain, newGain, numSamples);
+            SampleUtil::copyWithRampingGain(pIntermediateInput, pIn, oldGain, newGain, numSamples);
         }
 
         CSAMPLE* pIntermediateOutput;
-        for (EngineEffectRack* pRack : racks) {
-            if (pRack != nullptr) {
+        for (EngineEffectChain* pChain : chains) {
+            if (pChain) {
                 // Select an unused intermediate buffer for the next output
                 if (pIntermediateInput == m_buffer1.data()) {
                     pIntermediateOutput = m_buffer2.data();
                 } else {
                     pIntermediateOutput = m_buffer1.data();
                 }
 
-                if (pRack->process(inputHandle, outputHandle,
-                                   pIntermediateInput, pIntermediateOutput,
-                                   numSamples, sampleRate, groupFeatures)) {
-                    // Output of this rack becomes the input of the next rack.
+                if (pChain->process(inputHandle,
+                            outputHandle,
+                            pIntermediateInput,
+                            pIntermediateOutput,
+                            numSamples,
+                            sampleRate,
+                            groupFeatures)) {
+                    // Output of this chain becomes the input of the next chain.
                     pIntermediateInput = pIntermediateOutput;
                 }
             }
         }
-        // pIntermediateInput is the output of the last processed rack. It would be the
-        // intermediate input of the next rack if there was one.
+        // pIntermediateInput is the output of the last processed chain. It would
+        // be the intermediate input of the next chain if there was one.
         SampleUtil::add(pOut, pIntermediateInput, numSamples);
     }
 }
 
-bool EngineEffectsManager::addEffectRack(EngineEffectRack* pRack,
+bool EngineEffectsManager::addEffectChain(EngineEffectChain* pChain,
         SignalProcessingStage stage) {
-    QList<EngineEffectRack*>& rackList = m_racksByStage[stage];
-    VERIFY_OR_DEBUG_ASSERT(!rackList.contains(pRack)) {
+    QList<EngineEffectChain*>& chains = m_chainsByStage[stage];
+    VERIFY_OR_DEBUG_ASSERT(!chains.contains(pChain)) {
         return false;
     }
-    rackList.append(pRack);
+    chains.append(pChain);
```

Comment:
Won't this allocate? if not, can you document why?

Reply:
https://doc.qt.io/qt-5/qlist.html#append
> If this list is not shared, this operation is typically very fast (amortized constant time), because QList preallocates extra space on both sides of its internal buffer to allow for fast growth at both ends of the list.

This function is only called when Mixxx is initialized anyway so there isn't a problem even if it does allocated.

Reply:
I added a comment: 3af6071